app/controllers/articles_controller.rb

```diff
@@ -1,6 +1,5 @@
 class ArticlesController < ApplicationController
-  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
-
+  before_action :authenticate_user!
```

Comment:
What does the `authenticate_user` method do?

Reply:
The `authenticate_user` method prevents the code flow from continuing until a login has been performed.